FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/DemoTasks/KeepAliveMQTTExample.c

```diff
@@ -156,11 +151,17 @@
 #define mqttexampleKEEP_ALIVE_DELAY                  ( pdMS_TO_TICKS( ( ( mqttexampleKEEP_ALIVE_TIMEOUT_SECONDS / 4 ) * 1000 ) ) )
 
 /**
- * @brief Delay between MQTT publishes. Note that the receive loop also has a
- * timeout, so the total time between publishes is the sum of the two delays. The
- * keep-alive delay is added here so the keep-alive timer callback executes.
+ * @brief Time to wait before calling #MQTT_ReceiveLoop.
+ *
+ * @note This delay is deliberately chosen so that the keep-alive timer callback
+ * is invoked if an expected control packet is not received within 2 iterations.
```

Comment:
The loop iteration delay is set to 1000 ms here, and the timer runs every `mqttexampleKEEP_ALIVE_DELAY` (3000 ms), right? Am I missing something? (2 * 1000) ≯ 3000.

Reply:
How'd you get 1000? Loop iteration delay here is 15/8 = 1.875 s. Timer runs every 15/4 = 3.75 s.

Reply:
15/8 = 1 and 15/4 = 3 in integer division

Reply:
Ahh good catch

Reply:
That works, though you could also guard against the possibility of someone changing it to another value by always rounding up (e.g. (15+7)/8 and (15+3)/4))

Reply:
Don't know why I didn't just do this earlier but I just multiplied the ticks by a factor.